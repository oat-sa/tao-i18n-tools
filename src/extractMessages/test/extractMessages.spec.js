/**
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; under version 2
 * of the License (non-upgradable).
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 *
 * Copyright (c) 2019 (original work) Open Assessment Technologies SA
 *
 */

const extractMessages = require('../extractMessages.js');
const fs = require('fs');
const path = require('path');

let jsFileContent;
let svelteFileContent;

beforeAll(() => {
    jsFileContent = fs.readFileSync(path.join(__dirname, 'dummy.js'), 'utf8');
    svelteFileContent = fs.readFileSync(path.join(__dirname, 'dummy.svelte'), 'utf8');
});

describe('API', () => {
    it('should exist', function() {
        expect(extractMessages).toBeDefined;
    });

    it('should be a function', function() {
        expect(extractMessages).toBeInstanceOf(Function);
    });
});

describe('extractMessages with JavaScript and Svelte files', () => {
    it('should return a set of stings', function() {
        const strings = extractMessages(jsFileContent, 'dummy.js');
        const strings2 = extractMessages(svelteFileContent, 'dummy.svelte');

        expect(strings).toBeInstanceOf(Set);
        expect(strings).toMatchSnapshot();
        expect(strings2).toBeInstanceOf(Set);
        expect(strings2).toMatchSnapshot();
    });

    it('extractMessages returns right number of messages after parsing files', function() {
        const strings = extractMessages(jsFileContent, 'dummy.js');
        const strings2 = extractMessages(svelteFileContent, 'dummy.svelte');

        expect(strings.size).toEqual(2); // Not 3 because duplicate strings are not generated
        expect(strings.size).toMatchSnapshot();
        expect(strings2.size).toEqual(4);
        expect(strings2.size).toMatchSnapshot();
    });
});