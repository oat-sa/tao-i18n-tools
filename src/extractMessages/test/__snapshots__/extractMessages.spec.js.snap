// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extractMessages returns right number of messages with JavaScript and Svelte files should return a set of stings 1`] = `
Map {
  Object {
    "msgid": "Some dummy Â© string",
  } => Array [
    Object {
      "file": "dummy.js",
      "line": 3,
    },
  ],
  Object {
    "msgid": "duplicate string",
  } => Array [
    Object {
      "file": "dummy.js",
      "line": 4,
    },
    Object {
      "file": "dummy.js",
      "line": 5,
    },
  ],
  Object {
    "msgid": "Key",
    "msgid_plural": "Plural key",
  } => Array [
    Object {
      "file": "dummy.js",
      "line": 6,
    },
  ],
  Object {
    "msgid": "Duplicate key",
    "msgid_plural": "Duplicate plural key",
  } => Array [
    Object {
      "file": "dummy.js",
      "line": 7,
    },
    Object {
      "file": "dummy.js",
      "line": 8,
    },
  ],
}
`;

exports[`extractMessages returns right number of messages with JavaScript and Svelte files should return a set of stings 2`] = `
Map {
  Object {
    "msgid": "All work is done!",
  } => Array [
    Object {
      "file": "dummy.svelte",
      "line": 28,
    },
  ],
  Object {
    "msgid": "You have no more pending tasks for now!",
  } => Array [
    Object {
      "file": "dummy.svelte",
      "line": 29,
    },
  ],
  Object {
    "msgid": "Logout",
  } => Array [
    Object {
      "file": "dummy.svelte",
      "line": 30,
    },
  ],
  Object {
    "msgid": "Key",
    "msgid_plural": "Plural key",
  } => Array [
    Object {
      "file": "dummy.svelte",
      "line": 31,
    },
  ],
  Object {
    "msgid": "More dummy strings",
  } => Array [
    Object {
      "file": "dummy.svelte",
      "line": 2,
    },
  ],
}
`;
